numpy/_core/include/numpy/halffloat.h

```diff
@@ -1,6 +1,9 @@
 #ifndef NUMPY_CORE_INCLUDE_NUMPY_HALFFLOAT_H_
 #define NUMPY_CORE_INCLUDE_NUMPY_HALFFLOAT_H_
 
+/* Any file that includes Python.h must include it before any other files */
+/* https://docs.python.org/3/extending/extending.html#a-simple-example */
+/* npy_common.h includes Python.h so it also counts in this list */
```

Comment:
If we're going to touch every single header, I'd prefer it if this comment could be formatted like other comments in the numpy repo

```suggestion
/* 
   Any file that includes Python.h must include it before any other files
   https://docs.python.org/3/extending/extending.html#a-simple-example            
*/
```

I also think the addendum about `npy_common.h` is only relevant for files that include, and even then is maybe more confusing than helpful, since the most important thing is that `Python.h` comes first and the comment makes that clear.

Reply:
> I also think the addendum about `npy_common.h` is only relevant for files that include it,

The intent was that `npy_common.h` is a file that includes Python.h, and so either it or Python.h needs to be the first include in any file that includes it, but that's not relevant for this batch of comments and I can see how that wasn't clear.